Sources/Validators/CreditCardValidator.swift

```diff
@@ -0,0 +1,113 @@
+//
+//  CreditCardValidator.swift
+//  FormValidatorSwift
+//
+//  Created by Onur Ersel on 2016-11-02.
+//  Copyright © 2016 ustwo. All rights reserved.
+//
+
+import Foundation
+
+/**
+ *  The `CreditCardValidator` contains an `CreditCardCondition`. A valid string is a credit card number.
+ *  - seealso: `CreditCardCondition`
+ */
+public struct CreditCardValidator: Validator {
+    
+    
+    // MARK: - Properties
+    
+    public var conditions: [Condition]
+    
+    
+    // MARK: - Initializers
+    
+    /**
+     Initializes a `CreditCardValidator` with all card types.
+     */
+    public init() {
+        self.init(cardType: .all)
+    }
+    
+    /**
+     Initializes a `CreditCardValidator`.
+     - parameter cardType: Credit card type.
+     */
+    public init(cardType: CreditCardType) {
+        conditions = [CreditCardCondition(cardType: cardType)]
+    }
+    
+    
+    /**
+     Return card types valid with number given.
+    
+     - Parameters:
+      - creditCardNumber: Credit card number.
+      - completion: Completion callback, returns valid card types for given number.
+     */
+    public func cardTypeOf (creditCardNumber: String, completion : @escaping ([CreditCardType])->Void) {
+        
+        
+        DispatchQueue.global().async {
+            
+            var types = [CreditCardType]()
+            defer {
+                DispatchQueue.main.async {
+                    completion(types)
+                }
+            }
+            
+            if self.conditions.count == 0 || !(self.conditions[0] is CreditCardCondition) {
```

Comment:
You're right, this scope is quite a mess. Creating a separate condition for the checks seems like a more cleaner solution. 

Also not sure if should declare this function as static, since it's not dependent to the validator instance anyhow.